classes/session/SessionManager.php

```diff
@@ -15,106 +15,51 @@
+    private ?Session $userSession = null;
+
+    private PKPRequest $request;
 
     /**
      * Constructor.
      * Initialize session configuration and set PHP session handlers.
      * Attempts to rejoin a user's session if it exists, or create a new session otherwise.
      *
      */
-    public function __construct(SessionDAO $sessionDao, PKPRequest $request)
+    private function __construct(SessionDAO $sessionDao)
     {
         $this->sessionDao = $sessionDao;
+        $this->request = Application::get()->getRequest();
 
-        // Configure PHP session parameters
-        ini_set('session.use_trans_sid', 0);
-        ini_set('session.serialize_handler', 'php');
-        ini_set('session.use_cookies', 1);
-        ini_set('session.name', Config::getVar('general', 'session_cookie_name')); // Cookie name
-        ini_set('session.cookie_lifetime', 0);
-        ini_set('session.cookie_path', Config::getVar('general', 'session_cookie_path', $request->getBasePath() . '/'));
-        ini_set('session.cookie_domain', $request->getServerHost(null, false));
-        ini_set('session.cookie_httponly', 1);
-        ini_set('session.cookie_secure', Config::getVar('security', 'force_ssl'));
-        ini_set('session.gc_probability', 1);
-        ini_set('session.gc_maxlifetime', 60 * 60);
-        ini_set('session.cache_limiter', 'none');
-
-        session_set_save_handler(
-            [$this, 'open'],
-            [$this, 'close'],
-            [$this, 'read'],
-            [$this, 'write'],
-            [$this, 'destroy'],
-            [$this, 'gc']
-        );
-
-        // Initialize the session. This calls SessionManager::read() and
-        // sets $this->userSession if a session is present.
-        session_start();
-        $sessionId = session_id();
+        $this->configure();
+        $this->start();
 
-        $ip = $request->getRemoteAddr();
-        $userAgent = $request->getUserAgent();
-        $now = time();
-
-        // Check if the session is tied to the parent domain
-        $domain = $this->userSession ? $this->userSession->getDomain() : null;
-        if ($domain && $domain != $request->getServerHost(null, false)) {
-            // if current host contains . and the session domain (is a subdomain of the session domain), adjust the session's domain parameter to the parent
-            if (strtolower(substr($request->getServerHost(null, false), -1 - strlen($domain))) === '.' . strtolower($domain)) {
-                ini_set('session.cookie_domain', $domain);
+        // If there's a session assigned to the session ID
+        if ($this->userSession) {
+            // Validates it and refresh
+            if ($this->isValid($this->userSession)) {
+                $this->refresh();
+                return;
             }
+            // When invalid, regenerates the session ID without destroying the failed session (perhaps it belongs to another user)
+            session_regenerate_id();
```

Comment:
Here is the fix for the session fixation issue that I had, after failing the validation, it was destroying the session data at the database.